Add note on using transformations with filters (#5694)
segmentio/segment-docs: src/connections/destinations/destination-filters.md

Destination filters only filter events sent after filter setup. If you just added a destination filter but still see some events going through, you're likely seeing retries from failed events that occurred before you set up the filter.

When Segment sends an event to a destination but encounters a timeout error, it attempts to send the event again. As a result, if you add a destination filter while Segment is trying to send a failed event, these retries could filter through, since they reflect events that occurred before filter setup.

> #### How are Protocols Transformations handled by destination filters?
> 
> Events are first processed by a destination filter when a destination-specific Transformation is enabled. In contrast, for source-level Transformations, the transformation is applied to the events _before_ they reach the destination filter.

Reviewer: ```suggestion
#### How do destination filters handle Protocols Transformations?

When you enable a destination-specific Transformation, Segment processes it with a destination filter. Segment processes source-level Transformations before the events reach the destination filter.
```
